keps/sig-api-machinery/0017-apiserver-validates-etcd-instance.md

```diff
@@ -0,0 +1,61 @@
+---
+kep-number: 17
+title: apiserver validates etcd instance
+authors:
+  - "@ccding"
+owning-sig: sig-api-machinery
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-07-24
+last-updated: 2018-07-24
+status: provisional
+---
+
+# apiserver validates etcd instance
+
+## Summary
+
+We would like to have apiserver to validate the etcd instance to make sure it
+is connected to the proper one.
+
+## Motivation
+
+When there are multiple deployments of kubernetes and etcd, it may cause a
```

Comment:
Can you give more evidence to support this?  I can't say that I've heard this as a common complaint.  When would you have a Kubernetes apiserver that is confused about which etcd to talk to?

Reply:
Some examples of where this happened, what caused it, and any other issues or discussion in the community that indicates this is a common problem would be helpful for making a case for this change.

Reply:
Human makes mistakes. It happens when the apiserver is misconfigured, which messes up all the nodes--a disaster to the whole system. It happened a few times in our production environment.

Reply:
> It happened a few times in our production environment.

Securing etcd instances with per-cluster credentials would prevent apiservers messing with etcd instances belonging to other clusters.

That is a strong first recommendation (for many reasons) that is possible today, and would ensure the apiserver only talks to the intended etcd cluster.

Reply:
I'd support deprecating unsecured apiserver<->communication for this reason among others

Reply:
it makes sense. thanks